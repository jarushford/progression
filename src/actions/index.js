export const getTrainingData = (data) => ({
  type: 'GET_TRAINING',
  data
})

export const getProjects = (data) => ({
  type: 'GET_PROJECTS',
  data
})

export const clearProjects = () => ({
  type: 'CLEAR_PROJECTS'
})

export const getAscents = (data) => ({
  type: 'GET_ASCENTS',
  data
})

export const clearAscents = () => ({
  type: 'CLEAR_ASCENTS'
})

export const toggleDiscipline = (discipline) => ({
  type: 'TOGGLE_DISCIPLINE',
  discipline
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
})

export const toggleComplete = (key) => ({
  type: 'TOGGLE_COMPLETE',
  key
})
