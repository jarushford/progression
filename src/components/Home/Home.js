import React from 'react'
import '../../main.scss'

export default function Home({ toggleHomeScreen, boulder }) {
  let caption
  if (boulder) {
    caption = (
      <div className="caption">
        <h2>Terminus V12</h2>
        <h3>Bishop, CA</h3>
      </div>
    )
  } else {
    caption = (
      <div className="caption">
        <h2>Everything Is Karate 5.14d</h2>
        <h3>Bishop, CA</h3>
      </div>
    )
  }

  return (
    <div className="home">
      <div className="switcher">
        <input
          type="radio"
          name="style"
          id="sport"
          value={false}
          defaultChecked={!boulder}
          onClick={toggleHomeScreen}
        />
        <label htmlFor="sport" className="sport-label">sport</label>
        <input
          type="radio"
          name="style"
          id="boulder"
          value={true}
          defaultChecked={boulder}
          onClick={toggleHomeScreen}/>
        <label htmlFor="boulder" className="boulder-label">boulder</label>
      </div>
      {caption}
    </div>
  )
}