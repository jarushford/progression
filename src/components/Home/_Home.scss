.caption {
  position: absolute;
  bottom: 10px;
  left: 20px;
  color: $text;

  h2,
  h3 {
    margin: 5px 0;
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.9),
                 0 0 10px rgba(0, 0, 0, 0.9),
                 0 0 15px rgba(0, 0, 0, 0.9),
                 0 0 25px rgba(0, 0, 0, 0.9);
    font-weight: 400;
  }
}