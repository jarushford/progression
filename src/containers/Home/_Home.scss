.caption {
  position: absolute;
  bottom: 10px;
  left: 20px;
  color: $text;
  z-index: -1;

  h2,
  h3 {
    margin: 5px 0;
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.9),
                 0 0 10px rgba(0, 0, 0, 0.9),
                 0 0 15px rgba(0, 0, 0, 0.9),
                 0 0 25px rgba(0, 0, 0, 0.9);
    font-weight: 400;
  }
}

.switcher {
  float: right;
  margin: 100px 10px 0;
  background: $lightGrey;
  width: 150px;
  display: flex;
  border-radius: 10px;
  justify-content: space-between;
  box-shadow: inset 0 0 6px 2px rgba(0, 0, 0, 0.3);
  border: 2px solid $darkGrey;
}

input[type=radio] {
  display: none;
}

input[type=radio] + label {
  width: 100px;
  cursor: pointer;
  text-align: center;
  transition: .3s;
  padding: 5px;
  font-size: 1.2rem;
}

input[type=radio]:checked + label {
  cursor: pointer;
  background: $darkGrey;
  color: $text;
  transition: .3s;
  border-radius: 8px;
  box-shadow: inset 0 0 8px 2px rgba(200, 200, 200, 0.3);
}

input[type=radio]:checked + .sport-label {
  animation: slideLeft .2s ease;
}

input[type=radio]:checked + .boulder-label {
  animation: slideRight .2s ease;
}

@keyframes slideLeft {
  0% {
    transform: translateX(85px);
    color: transparent;
  }
  50% {
    color: transparent;
  }
  100% {
    transform: translateX(0);
    color: $text;
  }
}

@keyframes slideRight {
  0% {
    transform: translateX(-85px);
    color: transparent;
  }
  50% {
    color: transparent;
  }
  100% {
    transform: translateX(0);
    color: $text;
  }
}