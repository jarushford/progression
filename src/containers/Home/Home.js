import React from 'react'
import '../../main.scss'
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'
import * as ACTIONS from '../../actions/index'

function Home({ disciplineBoulder, toggleDiscipline }) {
  let caption
  if (disciplineBoulder) {
    caption = (
      <div className="caption">
        <h2>Jimmy Webb | Livin' Large | V15</h2>
        <h3>Rocklands, ZA</h3>
      </div>
    )
  } else {
    caption = (
      <div className="caption">
        <h2>Margo Hayes | La Rambla | 5.15a</h2>
        <h3>Siurana, ES</h3>
      </div>
    )
  }

  return (
    <div className="home">
      <div className="switcher">
        <input
          type="radio"
          name="style"
          id="sport"
          defaultChecked={!disciplineBoulder}
          onClick={() => toggleDiscipline(false)}
        />
        <label htmlFor="sport" className="sport-label">sport</label>
        <input
          type="radio"
          name="style"
          id="boulder"
          defaultChecked={disciplineBoulder}
          onClick={() => toggleDiscipline(true)}/>
        <label htmlFor="boulder" className="boulder-label">boulder</label>
      </div>
      {caption}
    </div>
  )
}

const mapStateToProps = (state) => ({
  disciplineBoulder: state.disciplineBoulder
})

const mapDispatchToProps = (dispatch) => ({
  toggleDiscipline: (isBoulder) => dispatch(ACTIONS.toggleDiscipline(isBoulder))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Home))