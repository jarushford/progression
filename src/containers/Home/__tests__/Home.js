import React from 'react'
import { shallow } from 'enzyme'
import { toggleDiscipline } from '../../../actions'
import { mapStateToProps, mapDispatchToProps, Home } from '../Home'

describe('Home', () => {
  let discipline
  let mockToggle

  beforeEach(() => {
    discipline = true
    mockToggle = jest.fn()
  })

  describe('Home Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Home disciplineBoulder={discipline} toggleDiscipline={mockToggle} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with discipline false', () => {
      const wrapper = shallow(<Home disciplineBoulder={!discipline} toggleDiscipline={mockToggle} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should toggle discipline when each input is clicked', () => {
      const wrapper = shallow(<Home disciplineBoulder={discipline} toggleDiscipline={mockToggle} />)

      wrapper.find('#sport').simulate('click')
      expect(mockToggle).toBeCalledWith(false)

      wrapper.find('#boulder').simulate('click')
      expect(mockToggle).toBeCalledWith(true)
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = { disciplineBoulder: true }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(mockState)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method toggleDiscipline', () => {
      const mockDispatch = jest.fn()
      const expected = toggleDiscipline(false)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleDiscipline(false)

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})