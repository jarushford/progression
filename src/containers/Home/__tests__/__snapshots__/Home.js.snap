// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    disciplineBoulder={true}
    toggleDiscipline={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="switcher"
        >
          <input
            defaultChecked={false}
            id="sport"
            name="style"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="sport-label"
            htmlFor="sport"
          >
            sport
          </label>
          <input
            defaultChecked={true}
            id="boulder"
            name="style"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="boulder-label"
            htmlFor="boulder"
          >
            boulder
          </label>
        </div>,
        <div
          className="caption"
        >
          <h2>
            Jimmy Webb | Livin' Large | V15
          </h2>
          <h3>
            Rocklands, ZA
          </h3>
        </div>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              defaultChecked={false}
              id="sport"
              name="style"
              onClick={[Function]}
              type="radio"
            />,
            <label
              className="sport-label"
              htmlFor="sport"
            >
              sport
            </label>,
            <input
              defaultChecked={true}
              id="boulder"
              name="style"
              onClick={[Function]}
              type="radio"
            />,
            <label
              className="boulder-label"
              htmlFor="boulder"
            >
              boulder
            </label>,
          ],
          "className": "switcher",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultChecked": false,
              "id": "sport",
              "name": "style",
              "onClick": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sport",
              "className": "sport-label",
              "htmlFor": "sport",
            },
            "ref": null,
            "rendered": "sport",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultChecked": true,
              "id": "boulder",
              "name": "style",
              "onClick": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "boulder",
              "className": "boulder-label",
              "htmlFor": "boulder",
            },
            "ref": null,
            "rendered": "boulder",
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Jimmy Webb | Livin' Large | V15
            </h2>,
            <h3>
              Rocklands, ZA
            </h3>,
          ],
          "className": "caption",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jimmy Webb | Livin' Large | V15",
            },
            "ref": null,
            "rendered": "Jimmy Webb | Livin' Large | V15",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rocklands, ZA",
            },
            "ref": null,
            "rendered": "Rocklands, ZA",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="switcher"
          >
            <input
              defaultChecked={false}
              id="sport"
              name="style"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="sport-label"
              htmlFor="sport"
            >
              sport
            </label>
            <input
              defaultChecked={true}
              id="boulder"
              name="style"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="boulder-label"
              htmlFor="boulder"
            >
              boulder
            </label>
          </div>,
          <div
            className="caption"
          >
            <h2>
              Jimmy Webb | Livin' Large | V15
            </h2>
            <h3>
              Rocklands, ZA
            </h3>
          </div>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                defaultChecked={false}
                id="sport"
                name="style"
                onClick={[Function]}
                type="radio"
              />,
              <label
                className="sport-label"
                htmlFor="sport"
              >
                sport
              </label>,
              <input
                defaultChecked={true}
                id="boulder"
                name="style"
                onClick={[Function]}
                type="radio"
              />,
              <label
                className="boulder-label"
                htmlFor="boulder"
              >
                boulder
              </label>,
            ],
            "className": "switcher",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultChecked": false,
                "id": "sport",
                "name": "style",
                "onClick": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sport",
                "className": "sport-label",
                "htmlFor": "sport",
              },
              "ref": null,
              "rendered": "sport",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultChecked": true,
                "id": "boulder",
                "name": "style",
                "onClick": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "boulder",
                "className": "boulder-label",
                "htmlFor": "boulder",
              },
              "ref": null,
              "rendered": "boulder",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Jimmy Webb | Livin' Large | V15
              </h2>,
              <h3>
                Rocklands, ZA
              </h3>,
            ],
            "className": "caption",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jimmy Webb | Livin' Large | V15",
              },
              "ref": null,
              "rendered": "Jimmy Webb | Livin' Large | V15",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rocklands, ZA",
              },
              "ref": null,
              "rendered": "Rocklands, ZA",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Home Home Component should match the snapshot with discipline false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    disciplineBoulder={false}
    toggleDiscipline={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="switcher"
        >
          <input
            defaultChecked={true}
            id="sport"
            name="style"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="sport-label"
            htmlFor="sport"
          >
            sport
          </label>
          <input
            defaultChecked={false}
            id="boulder"
            name="style"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="boulder-label"
            htmlFor="boulder"
          >
            boulder
          </label>
        </div>,
        <div
          className="caption"
        >
          <h2>
            Margo Hayes | La Rambla | 5.15a
          </h2>
          <h3>
            Siurana, ES
          </h3>
        </div>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              defaultChecked={true}
              id="sport"
              name="style"
              onClick={[Function]}
              type="radio"
            />,
            <label
              className="sport-label"
              htmlFor="sport"
            >
              sport
            </label>,
            <input
              defaultChecked={false}
              id="boulder"
              name="style"
              onClick={[Function]}
              type="radio"
            />,
            <label
              className="boulder-label"
              htmlFor="boulder"
            >
              boulder
            </label>,
          ],
          "className": "switcher",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultChecked": true,
              "id": "sport",
              "name": "style",
              "onClick": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sport",
              "className": "sport-label",
              "htmlFor": "sport",
            },
            "ref": null,
            "rendered": "sport",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultChecked": false,
              "id": "boulder",
              "name": "style",
              "onClick": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "boulder",
              "className": "boulder-label",
              "htmlFor": "boulder",
            },
            "ref": null,
            "rendered": "boulder",
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Margo Hayes | La Rambla | 5.15a
            </h2>,
            <h3>
              Siurana, ES
            </h3>,
          ],
          "className": "caption",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Margo Hayes | La Rambla | 5.15a",
            },
            "ref": null,
            "rendered": "Margo Hayes | La Rambla | 5.15a",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Siurana, ES",
            },
            "ref": null,
            "rendered": "Siurana, ES",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="switcher"
          >
            <input
              defaultChecked={true}
              id="sport"
              name="style"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="sport-label"
              htmlFor="sport"
            >
              sport
            </label>
            <input
              defaultChecked={false}
              id="boulder"
              name="style"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="boulder-label"
              htmlFor="boulder"
            >
              boulder
            </label>
          </div>,
          <div
            className="caption"
          >
            <h2>
              Margo Hayes | La Rambla | 5.15a
            </h2>
            <h3>
              Siurana, ES
            </h3>
          </div>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                defaultChecked={true}
                id="sport"
                name="style"
                onClick={[Function]}
                type="radio"
              />,
              <label
                className="sport-label"
                htmlFor="sport"
              >
                sport
              </label>,
              <input
                defaultChecked={false}
                id="boulder"
                name="style"
                onClick={[Function]}
                type="radio"
              />,
              <label
                className="boulder-label"
                htmlFor="boulder"
              >
                boulder
              </label>,
            ],
            "className": "switcher",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultChecked": true,
                "id": "sport",
                "name": "style",
                "onClick": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sport",
                "className": "sport-label",
                "htmlFor": "sport",
              },
              "ref": null,
              "rendered": "sport",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultChecked": false,
                "id": "boulder",
                "name": "style",
                "onClick": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "boulder",
                "className": "boulder-label",
                "htmlFor": "boulder",
              },
              "ref": null,
              "rendered": "boulder",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Margo Hayes | La Rambla | 5.15a
              </h2>,
              <h3>
                Siurana, ES
              </h3>,
            ],
            "className": "caption",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Margo Hayes | La Rambla | 5.15a",
              },
              "ref": null,
              "rendered": "Margo Hayes | La Rambla | 5.15a",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Siurana, ES",
              },
              "ref": null,
              "rendered": "Siurana, ES",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
